jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -5538,24 +5538,55 @@ static boolean permuteArgumentChecks(int[] permute, MethodType permuteType, Meth
 	}
 	
 	/**
-	 * Return the classData stored in the accessClass of the Lookup object.
+	 * Return the class data stored in the access class of the Lookup object.
 	 * 
-	 * @param caller Lookup object used to verify privileged access and retrieve classData.
-	 * @param unused.
-	 * @param type used to cast the classData of the accessClass.
+	 * @param caller Lookup object used to verify ORIGINAL access and retrieve class data.
+	 * @param name should match ConstantDescs.DEFAULT_NAME ("_").
+	 * @param type used to cast the class data of the access class.
 	 * 
-	 * @return the classData casted to the appropriate type.
-	 * @throws IllegalAccessException in the absence of full privilege access.
+	 * @return the class data casted to type if present; otherwise, null.
+	 * @throws IllegalArgumentException if name is not ConstantDescs.DEFAULT_NAME ("_").
+	 * @throws IllegalAccessException in the absence of ORIGINAL access.
+	 * @throws NullPointerException for null caller or type.
+	 * @throws ClassCastException if the class data cannot be converted to type.
 	 */
 	/*[IF JAVA_SPEC_VERSION >= 16]*/
 	public
 	/*[ENDIF] JAVA_SPEC_VERSION >= 16 */
-	static <T> T classData(Lookup caller, String unused, Class<T> type) throws IllegalAccessException {
-		if (caller.hasFullPrivilegeAccess()) {
-			Object classData = MethodHandleNatives.classData(caller.accessClass);
-			return type.cast(classData);
+	static <T> T classData(Lookup caller, String name, Class<T> type) throws IllegalAccessException {
+		if (!java.lang.constant.ConstantDescs.DEFAULT_NAME.equals(name)) {
+			throw new IllegalArgumentException();
+		}
+
+		if ((caller.lookupModes() & Lookup.ORIGINAL) != Lookup.ORIGINAL) {
+			throw new IllegalAccessException();
```

Comment:
```suggestion
			throw new IllegalAccessException("Lookup doesn't have ORIGINAL access bit");
```
This will help end users to figure out what's going on without having to ask for support.